src/category_theory/monoidal/braided.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.monoidal.functor
+
+/-!
+# Braided and symmetric monoidal categories
+
+The basic definitions of braided monoidal categories, and symmetric monoidal categories,
+as well as braided functors.
+
+## Implementation note
+
+We make `braided_monoidal_category` another typeclass, but then have `symmetric_monoidal_category`
+extend this. The rationale is that we are not carrying any additional data,
+just requiring a property.
+
+## Future work
+
+* Construct the Drinfeld center of a monoidal category as a braided monoidal category.
+* Say something about pseudo-natural transformations.
+
+-/
+
+open category_theory
+
+universes v v₁ v₂ v₃ u u₁ u₂ u₃
+
+namespace category_theory
+
+/--
+A braided monoidal category is a monoidal category equipped with a braiding isomorphism
+`β_ X Y : X ⊗ Y ≅ Y ⊗ X`
+which is natural in both arguments,
+and also satisfies the two hexagon identities.
+-/
+class braided_category (C : Type u) [category.{v} C] [monoidal_category.{v} C] :=
+-- braiding natural iso:
+(braiding             : Π X Y : C, X ⊗ Y ≅ Y ⊗ X)
+(braiding_naturality' : ∀ {X X' Y Y' : C} (f : X ⟶ Y) (g : X' ⟶ Y'),
+  (f ⊗ g) ≫ (braiding Y Y').hom = (braiding X X').hom ≫ (g ⊗ f) . obviously)
+-- hexagon identities:
+(hexagon_forward'     : Π X Y Z : C,
+    (α_ X Y Z).hom ≫ (braiding X (Y ⊗ Z)).hom ≫ (α_ Y Z X).hom
+  = ((braiding X Y).hom ⊗ (𝟙 Z)) ≫ (α_ Y X Z).hom ≫ ((𝟙 Y) ⊗ (braiding X Z).hom)
+  . obviously)
+(hexagon_reverse'     : Π X Y Z : C,
+    (α_ X Y Z).inv ≫ (braiding (X ⊗ Y) Z).hom ≫ (α_ Z X Y).inv
+  = ((𝟙 X) ⊗ (braiding Y Z).hom) ≫ (α_ X Z Y).inv ≫ ((braiding X Z).hom ⊗ (𝟙 Y))
+  . obviously)
+
+restate_axiom braided_category.braiding_naturality'
+attribute [simp] braided_category.braiding_naturality
```

Comment:
mark these `reassoc` too? and make the `reassoc`-ed version `simp` too, probably

Reply:
Done!

I also removed `simp` from the hexagon identities. We certainly weren't relying on these, and I think it may be better to wait for further applications before deciding on the best simp normal form. Likely we should reverse both hexagon equations, on the principle that the braidings "weigh more" than associators for simp normal form, but I think it's okay to wait here until someone actually wants `simp` to apply facts like this.